source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,258 @@
+method that specifies the criteria of the documents you want to retrieve.
+
+You can use a **cursor** to iterate over the documents returned by the ``Find()``
+method. A cursor is a mechanism that allows an application to iterate over database 
+results while holding only a subset of them in memory at a given time. Cursors
+are useful when your ``Find()`` method returns a large amount of documents.
+
+To use a cursor to iterate over the documents, pass a 
+query filter to the ``Find()`` method that specifies the criteria of the documents 
+you want to find, then chain the ``ToCursor()`` or ``ToCursorAsync()`` method.
+To view a synchronous or asynchronous example, select the corresponding tab. 
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: find-cursor-async
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var restaurants = await _restaurantsCollection.Find(filter).ToCursorAsync();
+
+   .. tab:: Synchronous
+      :tabid: find-cursor-sync
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var restaurants = _restaurantsCollection.Find(filter).ToCursor();
+
+If you are returning a small number of documents, or need your results returned
+as a ``List`` object, use the ``ToList()`` or ``ToListAsync()`` methods.
+
+To find multiple documents in a collection and hold them in memory as a list, pass a query filter 
+to the ``Find()`` method that specifies the criteria of the documents you want 
+to find, then chain the ``ToList()`` or ``ToListAsync()`` method.To view a
+synchronous or asynchronous example, select the corresponding tab.
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: find-list-async
+
+      .. code-block:: csharp
+
+         var restaurants = await _restaurantsCollection.Find(filter).ToList();
+
+   .. tab:: Synchronous
+      :tabid: find-list-sync
+
+      .. code-block:: csharp
+
+         var restaurants = _restaurantsCollection.Find(filter).ToListAsync();
+
+To see a full example of using the ``Find()`` method to find multiple documents,
+see :ref:`csharp-retrieve-additional-information`.
+
+.. note:: Find All Documents
+
+   To find all documents in a collection, pass an empty ``BsonDocument`` as a 
+   query filter to the ``Find()`` method.
+
+   To see a fully runnable example of using the ``Find()`` method to find all documents, see 
+   :ref:`csharp-retrieve-additional-information`.   
+
+Modify Find Behavior
+~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of the ``Find()`` method by passing
+a ``FindOptions`` object.
+
+You can configure the commonly used options with the following methods:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``BatchSize`` 
+     - | Gets or sets the number of documents to hold in a cursor at a given time.
+
+   * - ``Collation`` 
+     - | Sets the collation options. 
+
+   * - ``Comment`` 
+     - | Sets the comment to the query. To learn more about query comments, 
```

Comment:
NOTE: the linked page apparently doesn't contain all required information, for example which helpers support `$comment` and starting from what server version. It looks like it's supported everywhere (?) only starting from server 4.4. I was able to find a prove for db level only [here](https://www.mongodb.com/docs/manual/release-notes/4.4/#comment-option-available-to-all-database-commands) , but some xml comments [here](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#id23) say that in 4.4 we support all formats, but in the pre 4.4 servers sometimes we support only a comment in a string form, sometimes we don't support it at all. I would say we need a better documentation here. See this [ticket](https://jira.mongodb.org/browse/CSHARP-4471) for the context. It was hard to lead a user to the right page with all required details.

It might be a different ticket

Reply:
If I'm understanding correctly it sounds like this issue/confusion would all be addressed in the Server Manual documentation rather than through our C# docs directly, correct? In that case it would be a different ticket to the server docs team. 

Would it be less confusing to not link to that page at all or do you think it's still worthwhile to link to it here?